File: 프로그래머스/3/42628. 이중우선순위큐/이중우선순위큐.py
def solution(operations):
    answer = []
    queue = []
    
    for operation in operations:
        op = operation.split(" ")

        if op[0] == 'I':
            queue.append(int(op[1]))
        else:
            if len(queue):
                if op[1] == '-1':
                    queue.remove(min(queue)) 
                else:
                    queue.remove(max(queue)) 
                
    if len(queue) == 0:
        return [0,0]
    else:
        answer = [max(queue),min(queue)]
    return answer

File: 프로그래머스/3/42628. 이중우선순위큐/이중우선순위큐.js
function solution(operations) {
    var answer = [];
    operations.map((value)=>{
        let [op,num] = value.split(' ');
        num = Number(num);
        if(op==='I'){
            answer.push(num);
        } else if(num===1){
            let max = Math.max(...answer);
            answer.splice(answer.indexOf(max),1);
        }else{
            let min = Math.min(...answer);
            answer.splice(answer.indexOf(min),1);
        }
    });
    
    if (answer.length) return [Math.max(...answer),Math.min(...answer)];
    return [0,0];
}